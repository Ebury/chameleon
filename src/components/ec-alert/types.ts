export enum PropsAlertType {
  TEXT = 'text',
  DATE = 'date',
  NUMBER = 'number',
  TEL = 'tel'
}

export enum AlertType {
    ERROR = 'error', 
    INFO = 'info', 
    SUCCESS = 'success', 
    WARNING = 'warning'
}

export enum AlertEvent {
  UPDATE_OPEN = 'update:open',
  ACTION = 'action',
  CHANGE = 'change',
}

export interface AlertEvents {
  [AlertEvent.UPDATE_OPEN]: AlertProps['open']
  [AlertEvent.ACTION]: undefined
  [AlertEvent.CHANGE]: AlertProps['open']
}

export interface AlertProps {
  propsType?: PropsAlertType,
  type: AlertType,
  title: string,
  subtitle?: string,
  dismissable?: boolean,
  buttonText?: string,
  open?: boolean,
  responsive?: boolean
}
