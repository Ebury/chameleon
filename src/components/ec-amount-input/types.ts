import type { InputFieldProps } from '../ec-input-field/types';

export interface AmountInputProps extends Omit<InputFieldProps, 'modelValue'> {
  modelValue?: number | string | null,
  isMasked?: boolean,
  locale?: string,
  currency?: string,
}
