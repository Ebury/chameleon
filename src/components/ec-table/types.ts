import type { TableHeadColumn, TableHeadSort } from '../ec-table-head/types';
import { StickyColumnPosition } from '../ec-table-head/types';

export type { TableHeadColumn, TableHeadSort };
export { StickyColumnPosition };

export interface TableProps<TRow extends unknown[]> {
  columns?: TableHeadColumn[],
  sorts?: TableHeadSort[],
  data?: TRow[],
  totalRecords?: number,
  maxHeight?: string,
  stickyColumn?: StickyColumnPosition,
  title?: string,
  isCustomRowShown?: boolean,
  isTableHeaderHidden?: boolean
}
