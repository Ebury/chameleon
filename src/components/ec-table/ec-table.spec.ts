import { type ComponentMountingOptions, mount } from '@vue/test-utils';
import { vi } from 'vitest';
import { defineComponent, h } from 'vue';

import { SortDirection } from '../../enums';
import { StickyColumnPosition } from '../ec-table-head/types';
import EcTable from './ec-table.vue';
import type { TableProps } from './types';

function mountEcTable<TRow extends unknown[]>(props?: Partial<TableProps<TRow>>, mountOpts?: ComponentMountingOptions<typeof EcTable>) {
  return mount(EcTable, {
    props: {
      columns: [
        {
          name: 'lorem',
          title: 'Lorem',
        },
        {
          name: 'ipsum',
          title: 'Ipsum',
        },
      ],
      data: [
        ['foo', 'bar'],
        ['widgets', 'doodads'],
      ],
      ...props,
    },
    ...mountOpts,
  });
}

function mountTableAsTemplate<TRow extends unknown[]>(template: string, props?: Partial<TableProps<TRow>>, wrapperComponentOpts?: Record<string, unknown>, mountOpts?: ComponentMountingOptions<TableProps<TRow>>) {
  const Component = defineComponent({
    components: { EcTable },
    template,
    ...wrapperComponentOpts,
  });

  return mount(Component, {
    props,
    ...mountOpts,
  });
}

describe('EcTable', () => {
  it('should not render if no props are supplied', () => {
    const wrapper = mountEcTable({}, { props: {} });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should not render if not provided with any data', () => {
    const wrapper = mountEcTable({
      columns: undefined,
      data: undefined,
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should not render if not provided with an empty array of data', () => {
    const wrapper = mountEcTable({
      columns: undefined,
      data: [],
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render as expected if provided with data but no columns', () => {
    const wrapper = mountEcTable({
      columns: undefined,
      data: [
        ['foo', 'bar'],
        ['widgets', 'doodads'],
      ],
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render as expected if provided with data but no columns, and footer switched on', () => {
    const wrapper = mountEcTable({
      columns: undefined,
      data: [
        ['foo', 'bar'],
        ['widgets', 'doodads'],
      ],
    }, {
      slots: {
        footer: '<p>Random text</p>',
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render as expected if provided with data and columns', () => {
    const wrapper = mountEcTable();

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should have a column align to center if its type is icon', () => {
    const wrapper = mountTableAsTemplate(
      '<ec-table :columns="columns" :data="data"/>',
      {},
      {
        data() {
          return {
            columns: [
              {
                name: 'lorem',
                title: 'Lorem',
              },
              {
                name: 'ipsum',
                title: 'Ipsum',
                type: 'icon',
              },
            ],
            data: [
              ['foo', 'bar'],
              ['widgets', 'doodads'],
            ],
          };
        },
      },
    );

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should have a column align to the right if its type is currency', () => {
    const wrapper = mountTableAsTemplate(
      '<ec-table :columns="columns" :data="data"/>',
      {},
      {
        data() {
          return {
            columns: [
              {
                name: 'lorem',
                title: 'Lorem',
              },
              {
                name: 'ipsum',
                title: 'Ipsum',
                type: 'currency',
              },
            ],
            data: [
              ['foo', 'bar'],
              ['widgets', 'doodads'],
            ],
          };
        },
      },
    );

    expect(wrapper.element).toMatchSnapshot();
  });

  it('the tr should have class ec-table-row--is-clickable if we set the listener on row-click', () => {
    const onRowClick = vi.fn();
    const wrapper = mountEcTable(
      {
        columns: [
          {
            name: 'lorem',
            title: 'Lorem',
          },
          {
            name: 'ipsum',
            title: 'Ipsum',
            type: 'icon',
          },
        ],
        data: [
          ['foo', 'bar'],
          ['widgets', 'doodads'],
        ],
      },
      {
        attrs: {
          onRowClick,
        },
      },
    );

    const firstRow = wrapper.findByDataTest('ec-table__row--0');
    expect(firstRow.classes('ec-table__row--is-clickable')).toBe(true);
  });

  it('should render as expected if provided with data and columns, with footer switched on', () => {
    const wrapper = mountEcTable({}, {
      slots: {
        footer: '<p>Random text</p>',
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render as expected if provided with empty row and no columns, with footer switched on', () => {
    const wrapper = mountEcTable({
      columns: [],
      data: [[]],
    }, {
      slots: {
        footer: '<p>Random text</p>',
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render as expected if provided with rows and columns, with footer switched on and the title given', () => {
    const wrapper = mountEcTable({
      title: 'Title example',
    }, {
      slots: {
        footer: '<p>Random text</p>',
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render slots as expected', () => {
    const wrapper = mountEcTable({
      columns: [
        {
          name: 'lorem',
          title: 'Lorem',
        },
        {
          name: 'ipsum',
          title: 'Ipsum',
        },
      ],
      data: [
        ['foo', 'bar'],
        ['widgets', 'doodads'],
      ],
    }, {
      slots: {
        col2: ({ content }: { content: string }) => h('p', content),
        footer: '<p>Random text</p>',
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render custom row if window width is lower than 768px', () => {
    window.matchMedia = vi.fn().mockImplementation(query => ({
      matches: query === '(max-width: 767px)',
      media: '',
      onchange: null,
      addListener: vi.fn(),
      removeListener: vi.fn(),
    }));

    const wrapper = mountEcTable({
      columns: [
        {
          name: 'lorem',
          title: 'Lorem',
        },
        {
          name: 'ipsum',
          title: 'Ipsum',
        },
      ],
      data: [
        ['foo', 'bar'],
        ['widgets', 'doodads'],
      ],
    }, {
      slots: {
        default: '<p>Custom row</p>',
        col2: '<p>Column</p>',
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should only render custom row if "isCustomRowShown" is true', () => {
    const wrapper = mountEcTable({
      columns: [
        {
          name: 'lorem',
          title: 'Lorem',
        },
        {
          name: 'ipsum',
          title: 'Ipsum',
        },
      ],
      data: [
        ['foo', 'bar'],
        ['widgets', 'doodads'],
      ],
      isCustomRowShown: true,
    }, {
      slots: {
        default: '<p>Custom row</p>',
        col2: '<p>Column</p>',
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render sorting as expected', () => {
    const wrapper = mountEcTable({
      sorts: [
        { column: 'lorem', direction: SortDirection.ASC },
        { column: 'ipsum', direction: SortDirection.DESC },
      ],
      columns: [
        {
          name: 'lorem',
          title: 'Lorem',
          sortable: true,
        },
        {
          name: 'ipsum',
          title: 'Ipsum',
          sortable: true,
        },
      ],
    });

    expect(wrapper.findByDataTest('ec-table-head').element).toMatchSnapshot();
  });

  it('should notify parent about sorting', () => {
    const wrapper = mountEcTable({
      columns: [
        {
          name: 'lorem',
          title: 'Lorem',
          sortable: true,
        },
        {
          name: 'ipsum',
          title: 'Ipsum',
          sortable: true,
        },
      ],
    });
    wrapper.findByDataTest('ec-table-head__cell--0').findByDataTest('ec-table-sort__icon').trigger('click');
    expect(wrapper.emitted('sort')).toEqual([[{ name: 'lorem', sortable: true, title: 'Lorem' }]]);
  });

  it('the first th should have the ec-table-head__cell--sticky-left class if stickyColumn prop is left and the changes to right when the prop is get changed to right', async () => {
    const wrapper = mountEcTable({ stickyColumn: StickyColumnPosition.LEFT });

    expect(wrapper.findByDataTest('ec-table__cell--0').classes('ec-table__cell--sticky-left')).toBe(true);
    expect(wrapper.findByDataTest('ec-table__cell--1').classes('ec-table__cell--sticky-right')).toBe(false);
    await wrapper.setProps({ stickyColumn: 'right' });
    expect(wrapper.findByDataTest('ec-table__cell--1').classes('ec-table__cell--sticky-right')).toBe(true);
    expect(wrapper.findByDataTest('ec-table__cell--0').classes('ec-table__cell--sticky-left')).toBe(false);
  });

  it('should have the style height the same as given on the prop', () => {
    const wrapper = mountEcTable({ maxHeight: '400px' });

    expect(wrapper.findByDataTest('ec-table-scroll-container').attributes('style')).toBe('max-height: 400px;');
  });

  it('should emit the row-click event when you click on some row', async () => {
    const onRowClick = vi.fn();
    const wrapper = mountEcTable({}, {
      attrs: {
        onRowClick,
      },
    });

    await wrapper.findByDataTest('ec-table__row--0').trigger('click');
    expect(onRowClick.mock.calls[0]).toEqual([{ data: ['foo', 'bar'], rowIndex: 0 }]);
    await wrapper.findByDataTest('ec-table__row--1').trigger('click');
    expect(onRowClick.mock.calls[1]).toEqual([{ data: ['widgets', 'doodads'], rowIndex: 1 }]);
  });

  it('should render the style with the min-width on each cell of the column that have the prop given', () => {
    const columns = [
      {
        name: 'lorem',
        title: 'Lorem',
        minWidth: '250px',
      },
      {
        name: 'ipsum',
        title: 'Ipsum',
      },
    ];
    const wrapper = mountEcTable({ columns });
    const wrapperArray = wrapper.findAllByDataTest('ec-table__cell--0');
    expect(wrapperArray.length).toBe(columns.length);
    wrapperArray.forEach(wrapperItem => expect(wrapperItem.attributes('style')).toBe('min-width: 250px;'));
  });

  it('should render the style with the max-width on each cell of the column that have the prop given', () => {
    const columns = [
      {
        name: 'lorem',
        title: 'Lorem',
        maxWidth: '250px',
      },
      {
        name: 'ipsum',
        title: 'Ipsum',
      },
    ];
    const wrapper = mountEcTable({ columns });

    expect(wrapper.element).toMatchSnapshot();
  });
});
