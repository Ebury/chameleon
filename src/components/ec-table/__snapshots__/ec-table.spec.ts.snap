// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EcTable > should have a column align to center if its type is icon 1`] = `
<div>
  <!--v-if-->
  <div
    class="ec-table-scroll-container"
    data-test="ec-table-scroll-container"
  >
    <table
      class="ec-table"
    >
      <thead
        class="ec-table-head"
        data-test="ec-table-head"
      >
        <tr>
          
          <th
            class="ec-table-head__cell"
            data-test="ec-table-head__cell ec-table-head__cell--0"
            scope="col"
          >
            <span
              class="ec-table-head__cell-wrapper"
            >
              <span
                class="ec-table-head__cell-text"
              >
                Lorem
              </span>
              <!--v-if-->
              <!--v-if-->
            </span>
          </th>
          <th
            class="ec-table-head__cell"
            data-test="ec-table-head__cell ec-table-head__cell--1"
            scope="col"
          >
            <span
              class="ec-table-head__cell-wrapper ec-table-head__cell-wrapper--is-type-icon"
            >
              <span
                class="ec-table-head__cell-text"
              >
                Ipsum
              </span>
              <!--v-if-->
              <!--v-if-->
            </span>
          </th>
          
        </tr>
      </thead>
      <tbody>
        
        <tr
          class=""
          data-test="ec-table__row ec-table__row--0"
        >
          
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--0"
          >
            
            foo
            
          </td>
          <td
            class="ec-table__cell ec-table__cell--is-type-icon"
            data-test="ec-table__cell ec-table__cell--1"
          >
            
            bar
            
          </td>
          
        </tr>
        <tr
          class=""
          data-test="ec-table__row ec-table__row--1"
        >
          
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--0"
          >
            
            widgets
            
          </td>
          <td
            class="ec-table__cell ec-table__cell--is-type-icon"
            data-test="ec-table__cell ec-table__cell--1"
          >
            
            doodads
            
          </td>
          
        </tr>
        
      </tbody>
      <!--v-if-->
    </table>
  </div>
</div>
`;

exports[`EcTable > should have a column align to the right if its type is currency 1`] = `
<div>
  <!--v-if-->
  <div
    class="ec-table-scroll-container"
    data-test="ec-table-scroll-container"
  >
    <table
      class="ec-table"
    >
      <thead
        class="ec-table-head"
        data-test="ec-table-head"
      >
        <tr>
          
          <th
            class="ec-table-head__cell"
            data-test="ec-table-head__cell ec-table-head__cell--0"
            scope="col"
          >
            <span
              class="ec-table-head__cell-wrapper"
            >
              <span
                class="ec-table-head__cell-text"
              >
                Lorem
              </span>
              <!--v-if-->
              <!--v-if-->
            </span>
          </th>
          <th
            class="ec-table-head__cell"
            data-test="ec-table-head__cell ec-table-head__cell--1"
            scope="col"
          >
            <span
              class="ec-table-head__cell-wrapper ec-table-head__cell-wrapper--is-type-currency"
            >
              <span
                class="ec-table-head__cell-text"
              >
                Ipsum
              </span>
              <!--v-if-->
              <!--v-if-->
            </span>
          </th>
          
        </tr>
      </thead>
      <tbody>
        
        <tr
          class=""
          data-test="ec-table__row ec-table__row--0"
        >
          
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--0"
          >
            
            foo
            
          </td>
          <td
            class="ec-table__cell ec-table__cell--is-type-currency"
            data-test="ec-table__cell ec-table__cell--1"
          >
            
            bar
            
          </td>
          
        </tr>
        <tr
          class=""
          data-test="ec-table__row ec-table__row--1"
        >
          
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--0"
          >
            
            widgets
            
          </td>
          <td
            class="ec-table__cell ec-table__cell--is-type-currency"
            data-test="ec-table__cell ec-table__cell--1"
          >
            
            doodads
            
          </td>
          
        </tr>
        
      </tbody>
      <!--v-if-->
    </table>
  </div>
</div>
`;

exports[`EcTable > should not render if no props are supplied 1`] = `<!--v-if-->`;

exports[`EcTable > should not render if not provided with an empty array of data 1`] = `<!--v-if-->`;

exports[`EcTable > should not render if not provided with any data 1`] = `<!--v-if-->`;

exports[`EcTable > should only render custom row if "isCustomRowShown" is true 1`] = `
<div>
  <!--v-if-->
  <div
    class="ec-table-scroll-container"
    data-test="ec-table-scroll-container"
  >
    <table
      class="ec-table tw-table-fixed"
    >
      <!--v-if-->
      <tbody>
        
        <tr
          class=""
          data-test="ec-table__row ec-table__row--0"
        >
          <td
            colspan="2"
          >
            
            <p>
              Custom row
            </p>
            
          </td>
        </tr>
        <tr
          class=""
          data-test="ec-table__row ec-table__row--1"
        >
          <td
            colspan="2"
          >
            
            <p>
              Custom row
            </p>
            
          </td>
        </tr>
        
      </tbody>
      <!--v-if-->
    </table>
  </div>
</div>
`;

exports[`EcTable > should render as expected if provided with data and columns 1`] = `
<div>
  <!--v-if-->
  <div
    class="ec-table-scroll-container"
    data-test="ec-table-scroll-container"
  >
    <table
      class="ec-table"
    >
      <thead
        class="ec-table-head"
        data-test="ec-table-head"
      >
        <tr>
          
          <th
            class="ec-table-head__cell"
            data-test="ec-table-head__cell ec-table-head__cell--0"
            scope="col"
          >
            <span
              class="ec-table-head__cell-wrapper"
            >
              <span
                class="ec-table-head__cell-text"
              >
                Lorem
              </span>
              <!--v-if-->
              <!--v-if-->
            </span>
          </th>
          <th
            class="ec-table-head__cell"
            data-test="ec-table-head__cell ec-table-head__cell--1"
            scope="col"
          >
            <span
              class="ec-table-head__cell-wrapper"
            >
              <span
                class="ec-table-head__cell-text"
              >
                Ipsum
              </span>
              <!--v-if-->
              <!--v-if-->
            </span>
          </th>
          
        </tr>
      </thead>
      <tbody>
        
        <tr
          class=""
          data-test="ec-table__row ec-table__row--0"
        >
          
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--0"
          >
            
            foo
            
          </td>
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--1"
          >
            
            bar
            
          </td>
          
        </tr>
        <tr
          class=""
          data-test="ec-table__row ec-table__row--1"
        >
          
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--0"
          >
            
            widgets
            
          </td>
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--1"
          >
            
            doodads
            
          </td>
          
        </tr>
        
      </tbody>
      <!--v-if-->
    </table>
  </div>
</div>
`;

exports[`EcTable > should render as expected if provided with data and columns, with footer switched on 1`] = `
<div>
  <!--v-if-->
  <div
    class="ec-table-scroll-container"
    data-test="ec-table-scroll-container"
  >
    <table
      class="ec-table"
    >
      <thead
        class="ec-table-head"
        data-test="ec-table-head"
      >
        <tr>
          
          <th
            class="ec-table-head__cell"
            data-test="ec-table-head__cell ec-table-head__cell--0"
            scope="col"
          >
            <span
              class="ec-table-head__cell-wrapper"
            >
              <span
                class="ec-table-head__cell-text"
              >
                Lorem
              </span>
              <!--v-if-->
              <!--v-if-->
            </span>
          </th>
          <th
            class="ec-table-head__cell"
            data-test="ec-table-head__cell ec-table-head__cell--1"
            scope="col"
          >
            <span
              class="ec-table-head__cell-wrapper"
            >
              <span
                class="ec-table-head__cell-text"
              >
                Ipsum
              </span>
              <!--v-if-->
              <!--v-if-->
            </span>
          </th>
          
        </tr>
      </thead>
      <tbody>
        
        <tr
          class=""
          data-test="ec-table__row ec-table__row--0"
        >
          
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--0"
          >
            
            foo
            
          </td>
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--1"
          >
            
            bar
            
          </td>
          
        </tr>
        <tr
          class=""
          data-test="ec-table__row ec-table__row--1"
        >
          
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--0"
          >
            
            widgets
            
          </td>
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--1"
          >
            
            doodads
            
          </td>
          
        </tr>
        
      </tbody>
      <tfoot
        class="ec-table-footer"
        data-test="ec-table-footer"
      >
        <tr>
          <td
            class="ec-table-footer__cell"
            colspan="2"
          >
            <div
              class="ec-table-footer__contents"
            >
              
              
              <p>
                Random text
              </p>
              
              
            </div>
          </td>
        </tr>
      </tfoot>
    </table>
  </div>
</div>
`;

exports[`EcTable > should render as expected if provided with data but no columns 1`] = `
<div>
  <!--v-if-->
  <div
    class="ec-table-scroll-container"
    data-test="ec-table-scroll-container"
  >
    <table
      class="ec-table"
    >
      <!--v-if-->
      <tbody>
        
        <tr
          class=""
          data-test="ec-table__row ec-table__row--0"
        >
          
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--0"
          >
            
            foo
            
          </td>
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--1"
          >
            
            bar
            
          </td>
          
        </tr>
        <tr
          class=""
          data-test="ec-table__row ec-table__row--1"
        >
          
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--0"
          >
            
            widgets
            
          </td>
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--1"
          >
            
            doodads
            
          </td>
          
        </tr>
        
      </tbody>
      <!--v-if-->
    </table>
  </div>
</div>
`;

exports[`EcTable > should render as expected if provided with data but no columns, and footer switched on 1`] = `
<div>
  <!--v-if-->
  <div
    class="ec-table-scroll-container"
    data-test="ec-table-scroll-container"
  >
    <table
      class="ec-table"
    >
      <!--v-if-->
      <tbody>
        
        <tr
          class=""
          data-test="ec-table__row ec-table__row--0"
        >
          
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--0"
          >
            
            foo
            
          </td>
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--1"
          >
            
            bar
            
          </td>
          
        </tr>
        <tr
          class=""
          data-test="ec-table__row ec-table__row--1"
        >
          
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--0"
          >
            
            widgets
            
          </td>
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--1"
          >
            
            doodads
            
          </td>
          
        </tr>
        
      </tbody>
      <tfoot
        class="ec-table-footer"
        data-test="ec-table-footer"
      >
        <tr>
          <td
            class="ec-table-footer__cell"
            colspan="2"
          >
            <div
              class="ec-table-footer__contents"
            >
              
              
              <p>
                Random text
              </p>
              
              
            </div>
          </td>
        </tr>
      </tfoot>
    </table>
  </div>
</div>
`;

exports[`EcTable > should render as expected if provided with empty row and no columns, with footer switched on 1`] = `
<div>
  <!--v-if-->
  <div
    class="ec-table-scroll-container"
    data-test="ec-table-scroll-container"
  >
    <table
      class="ec-table"
    >
      <!--v-if-->
      <tbody>
        
        <tr
          class=""
          data-test="ec-table__row ec-table__row--0"
        >
          
          
        </tr>
        
      </tbody>
      <tfoot
        class="ec-table-footer"
        data-test="ec-table-footer"
      >
        <tr>
          <td
            class="ec-table-footer__cell"
          >
            <div
              class="ec-table-footer__contents"
            >
              
              
              <p>
                Random text
              </p>
              
              
            </div>
          </td>
        </tr>
      </tfoot>
    </table>
  </div>
</div>
`;

exports[`EcTable > should render as expected if provided with rows and columns, with footer switched on and the title given 1`] = `
<div>
  <div
    class="ec-table__title"
  >
    Title example
  </div>
  <div
    class="ec-table-scroll-container"
    data-test="ec-table-scroll-container"
  >
    <table
      aria-label="Title example"
      class="ec-table"
    >
      <thead
        class="ec-table-head"
        data-test="ec-table-head"
      >
        <tr>
          
          <th
            class="ec-table-head__cell"
            data-test="ec-table-head__cell ec-table-head__cell--0"
            scope="col"
          >
            <span
              class="ec-table-head__cell-wrapper"
            >
              <span
                class="ec-table-head__cell-text"
              >
                Lorem
              </span>
              <!--v-if-->
              <!--v-if-->
            </span>
          </th>
          <th
            class="ec-table-head__cell"
            data-test="ec-table-head__cell ec-table-head__cell--1"
            scope="col"
          >
            <span
              class="ec-table-head__cell-wrapper"
            >
              <span
                class="ec-table-head__cell-text"
              >
                Ipsum
              </span>
              <!--v-if-->
              <!--v-if-->
            </span>
          </th>
          
        </tr>
      </thead>
      <tbody>
        
        <tr
          class=""
          data-test="ec-table__row ec-table__row--0"
        >
          
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--0"
          >
            
            foo
            
          </td>
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--1"
          >
            
            bar
            
          </td>
          
        </tr>
        <tr
          class=""
          data-test="ec-table__row ec-table__row--1"
        >
          
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--0"
          >
            
            widgets
            
          </td>
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--1"
          >
            
            doodads
            
          </td>
          
        </tr>
        
      </tbody>
      <tfoot
        class="ec-table-footer"
        data-test="ec-table-footer"
      >
        <tr>
          <td
            class="ec-table-footer__cell"
            colspan="2"
          >
            <div
              class="ec-table-footer__contents"
            >
              
              
              <p>
                Random text
              </p>
              
              
            </div>
          </td>
        </tr>
      </tfoot>
    </table>
  </div>
</div>
`;

exports[`EcTable > should render custom row if window width is lower than 768px 1`] = `
<div>
  <!--v-if-->
  <div
    class="ec-table-scroll-container"
    data-test="ec-table-scroll-container"
  >
    <table
      class="ec-table tw-table-fixed"
    >
      <!--v-if-->
      <tbody>
        
        <tr
          class=""
          data-test="ec-table__row ec-table__row--0"
        >
          <td
            colspan="2"
          >
            
            <p>
              Custom row
            </p>
            
          </td>
        </tr>
        <tr
          class=""
          data-test="ec-table__row ec-table__row--1"
        >
          <td
            colspan="2"
          >
            
            <p>
              Custom row
            </p>
            
          </td>
        </tr>
        
      </tbody>
      <!--v-if-->
    </table>
  </div>
</div>
`;

exports[`EcTable > should render slots as expected 1`] = `
<div>
  <!--v-if-->
  <div
    class="ec-table-scroll-container"
    data-test="ec-table-scroll-container"
  >
    <table
      class="ec-table"
    >
      <thead
        class="ec-table-head"
        data-test="ec-table-head"
      >
        <tr>
          
          <th
            class="ec-table-head__cell"
            data-test="ec-table-head__cell ec-table-head__cell--0"
            scope="col"
          >
            <span
              class="ec-table-head__cell-wrapper"
            >
              <span
                class="ec-table-head__cell-text"
              >
                Lorem
              </span>
              <!--v-if-->
              <!--v-if-->
            </span>
          </th>
          <th
            class="ec-table-head__cell"
            data-test="ec-table-head__cell ec-table-head__cell--1"
            scope="col"
          >
            <span
              class="ec-table-head__cell-wrapper"
            >
              <span
                class="ec-table-head__cell-text"
              >
                Ipsum
              </span>
              <!--v-if-->
              <!--v-if-->
            </span>
          </th>
          
        </tr>
      </thead>
      <tbody>
        
        <tr
          class=""
          data-test="ec-table__row ec-table__row--0"
        >
          
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--0"
          >
            
            foo
            
          </td>
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--1"
          >
            
            <p>
              bar
            </p>
            
          </td>
          
        </tr>
        <tr
          class=""
          data-test="ec-table__row ec-table__row--1"
        >
          
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--0"
          >
            
            widgets
            
          </td>
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--1"
          >
            
            <p>
              doodads
            </p>
            
          </td>
          
        </tr>
        
      </tbody>
      <tfoot
        class="ec-table-footer"
        data-test="ec-table-footer"
      >
        <tr>
          <td
            class="ec-table-footer__cell"
            colspan="2"
          >
            <div
              class="ec-table-footer__contents"
            >
              
              
              <p>
                Random text
              </p>
              
              
            </div>
          </td>
        </tr>
      </tfoot>
    </table>
  </div>
</div>
`;

exports[`EcTable > should render sorting as expected 1`] = `
<thead
  class="ec-table-head"
  data-test="ec-table-head"
>
  <tr>
    
    <th
      class="ec-table-head__cell"
      data-test="ec-table-head__cell ec-table-head__cell--0"
      scope="col"
    >
      <span
        class="ec-table-head__cell-wrapper"
      >
        <span
          class="ec-table-head__cell-text"
        >
          Lorem
        </span>
        <!--v-if-->
        <a
          class="ec-table-sort ec-table-head__icon"
          data-test="ec-table-sort"
          href="#"
          title="Sorted ascending"
        >
          <svg
            class="ec-icon ec-table-sort__icon ec-table-sort__icon--asc"
            data-test="ec-table-sort__icon"
            height="16"
            width="16"
          >
            <use
              xlink:href="#ec-simple-arrow-drop-down"
            />
          </svg>
          <span
            class="ec-table-sort__text"
          >
            asc
          </span>
        </a>
      </span>
    </th>
    <th
      class="ec-table-head__cell"
      data-test="ec-table-head__cell ec-table-head__cell--1"
      scope="col"
    >
      <span
        class="ec-table-head__cell-wrapper"
      >
        <span
          class="ec-table-head__cell-text"
        >
          Ipsum
        </span>
        <!--v-if-->
        <a
          class="ec-table-sort ec-table-head__icon"
          data-test="ec-table-sort"
          href="#"
          title="Sorted descending"
        >
          <svg
            class="ec-icon ec-table-sort__icon ec-table-sort__icon--desc"
            data-test="ec-table-sort__icon"
            height="16"
            width="16"
          >
            <use
              xlink:href="#ec-simple-arrow-drop-down"
            />
          </svg>
          <span
            class="ec-table-sort__text"
          >
            desc
          </span>
        </a>
      </span>
    </th>
    
  </tr>
</thead>
`;

exports[`EcTable > should render the style with the max-width on each cell of the column that have the prop given 1`] = `
<div>
  <!--v-if-->
  <div
    class="ec-table-scroll-container"
    data-test="ec-table-scroll-container"
  >
    <table
      class="ec-table"
    >
      <thead
        class="ec-table-head"
        data-test="ec-table-head"
      >
        <tr>
          
          <th
            class="ec-table-head__cell"
            data-test="ec-table-head__cell ec-table-head__cell--0"
            scope="col"
            style="max-width: 250px;"
          >
            <span
              class="ec-table-head__cell-wrapper"
            >
              <span
                class="ec-table-head__cell-text ec-table-head__cell-text--has-max-width"
              >
                Lorem
              </span>
              <!--v-if-->
              <!--v-if-->
            </span>
          </th>
          <th
            class="ec-table-head__cell"
            data-test="ec-table-head__cell ec-table-head__cell--1"
            scope="col"
          >
            <span
              class="ec-table-head__cell-wrapper"
            >
              <span
                class="ec-table-head__cell-text"
              >
                Ipsum
              </span>
              <!--v-if-->
              <!--v-if-->
            </span>
          </th>
          
        </tr>
      </thead>
      <tbody>
        
        <tr
          class=""
          data-test="ec-table__row ec-table__row--0"
        >
          
          <td
            class="ec-table__cell ec-table__cell--has-max-width"
            data-test="ec-table__cell ec-table__cell--0"
            style="max-width: 250px;"
          >
            
            foo
            
          </td>
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--1"
          >
            
            bar
            
          </td>
          
        </tr>
        <tr
          class=""
          data-test="ec-table__row ec-table__row--1"
        >
          
          <td
            class="ec-table__cell ec-table__cell--has-max-width"
            data-test="ec-table__cell ec-table__cell--0"
            style="max-width: 250px;"
          >
            
            widgets
            
          </td>
          <td
            class="ec-table__cell"
            data-test="ec-table__cell ec-table__cell--1"
          >
            
            doodads
            
          </td>
          
        </tr>
        
      </tbody>
      <!--v-if-->
    </table>
  </div>
</div>
`;
